src/client/scripts/game/misc/enginegame.js

```diff
@@ -0,0 +1,96 @@
+
+// This module keeps track of the data of the engine game we are currently in.
+
+"use strict";
+
+const enginegame = (function(){
+
+    /** Whether we are currently in an engine game. */
+    let inEngineGame = false
+    let ourColor; // white/black
+
+    const engineTimeLimitPerMove = 1000
+
+    function areInEngineGame() { return inEngineGame }
+
+    function getOurColor() { return ourColor }
+
+    /**
+     * This has to be called before and separate from {@link initEngineGame}
+     * because loading the gamefile and the mesh generation requires this script to know our color.
+     * @param {string} color - The color we are in this engine game
+     */
+    function setColorAndGameID(gameOptions) {
+        inEngineGame = true
+        ourColor = gameOptions.youAreColor;
+    }
+
+    /**
+     * Inits an engine game according to the options provided
+     * @param {Object} gameOptions - An object that contains the property `youAreColor`
+     */
+    function initEngineGame (gameOptions) {
+        // These make sure it will place us in black's perspective
+        perspective.resetRotations()
```

Comment:
gameOptions redundant here. But okay, maybe this same issue is over in onlinegame.js as well